examples/cra-ts-kitchen-sink/tsconfig.json

```diff
@@ -12,10 +12,12 @@
     "strict": true,
     "forceConsistentCasingInFileNames": true,
     "resolveJsonModule": true,
-    "isolatedModules": true,
-    "noEmit": true
+    "isolatedModules": true
```

Comment:
using `isolatedModules` in all our tsconfigs could improve tsc performance

Reply:
should we also add `declaration: false` here?! I'm not sure what is the best way to make sure this one won't emit any typed files.  If we want an emission, then this have to be set to false.  See this issue on [TS repo here](https://github.com/Microsoft/TypeScript/issues/29490).

Reply:
`isolatedModules: true`
```real    0m18.465s
user    0m0.045s
sys     0m0.045s```

`isolatedModules: false`
real    0m18.345s
user    0m0.030s
sys     0m0.091s```

Why should it be faster? What does this option even do? I tried to find more accurate documentation than "exports every file as separate module"

Also keep in mind that this is an example app and depends on the decisions a project owner makes, not storybook.

@ndelangen you added `isolatedModules: true` - Were you just playing around with the options or was this on purpose?

Reply:
it could/should be faster because tsc doesn't need cross-file information to emit. Doesn't look like it makes any difference for us though. 

I would still recommend adding it in because it helps enforce "Babel's single file emit architecture". See the caveats section at https://devblogs.microsoft.com/typescript/typescript-and-babel-7/

Reply:
I just found out: ` TS5053: Option 'declaration' cannot be specified with option 'isolatedModules'.` 🐙

Reply:
That is why I suggest to add `declaration: false` together with `isolatedModules`, maybe with a comment?  In my own project, I have `tsc --isolatedModules --noEmit --watch` for type checking, and `tsc --declaration --removeComments --emitDeclarationOnly` for type emission. The rest of them goes through babel.

Reply:
I'm with @leoyli on this one, sounds like that's what we want?

Reply:
Hm question:
- You start `yarn dev` and `yarn dev:tsc` needs to create declaration files for other internal packages that rely on that package: How can I do `declaration: false` here
- You start `yarn bootstrap --core` and internally packages again rely on the `d.ts` files of other packages: How can I do `declaration: false`

Is there something I don't get or is this a misunderstanding? 🙂

Reply:
I mean, yes - it won't fail but at least your IDE will mark those imports red because they are missing - Isn't it confusing if you get red lines in your IDE?